plugins/org.eclipse.n4js.generator/src/org/eclipse/n4js/generator/N4JSCompositeGenerator.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2016 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.generator;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.n4js.N4JSGlobals;
+import org.eclipse.n4js.external.ExternalLibraryUriHelper;
+import org.eclipse.n4js.generator.common.CompilerDescriptor;
+import org.eclipse.n4js.generator.common.IComposedGenerator;
+import org.eclipse.n4js.generator.common.ISubGenerator;
+import org.eclipse.n4js.generator.common.SubgeneratorsRegistry;
+import org.eclipse.n4js.projectModel.IN4JSCore;
+import org.eclipse.n4js.projectModel.IN4JSSourceContainer;
+import org.eclipse.xtext.generator.IFileSystemAccess;
+
+import com.google.inject.Inject;
+
+/**
+ * A composite generator is responsible for a language. A composite generator delegates the generation logics to its
+ * children (subgenerators).
+ */
+public class N4JSCompositeGenerator implements IComposedGenerator {
```

Comment:
1) terminology mismatch between class and interface: composite vs. composed
2) probably `@Singleton` required. BUT: check if it is possible to add this to the interface and it will still be picked up by Guice; if that is possible -> add to interface.